monai/transforms/signal/array.py

```diff
@@ -0,0 +1,499 @@
+            signal: input 1 dimension signal to which gaussian noise will be added
+        """
+        self.randomize(None)
+        self.magnitude = self.R.uniform(low=self.boundaries[0], high=self.boundaries[1])
+        length = signal.shape[1]
+        gaussiannoise = self.magnitude * torch.randn(length)
+
+        signal = convert_to_tensor(signal) + gaussiannoise
+
+        return signal
+
+
+class SignalRandAddSquarePulsePartial(RandomizableTransform):
+    """
+    Add a random partial square pulse to a signal
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        boundaries: Sequence[float] = (0.01, 0.2),
+        frequencies: Sequence[float] = (0.001, 0.02),
+        fraction: Sequence[float] = (0.01, 0.2),
+    ) -> None:
+        """
+        Args:
+            boundaries: list defining lower and upper boundaries for the square pulse magnitude,
+                lower and upper values need to be positive , default : ``[0.01, 0.2]``
+            frequencies: list defining lower and upper frequencies for square pulse
+                signal generation example : ``[0.001, 0.02]``
+            fraction: list defining lower and upper boundaries for partial square pulse generation
+                default: ``[0.01, 0.2]``
+        """
+        super().__init__()
+        check_boundaries(boundaries)
+        self.boundaries = boundaries
+        self.frequencies = frequencies
+        self.fraction = fraction
+
+    def __call__(self, signal: NdarrayOrTensor) -> NdarrayOrTensor:
+        """
+        Args:
+            signal: input 1 dimension signal to which a partial square pulse will be added
+        """
+        self.randomize(None)
+        self.magnitude = self.R.uniform(low=self.boundaries[0], high=self.boundaries[1])
+        self.fracs = self.R.uniform(low=self.fraction[0], high=self.fraction[1])
+        self.freqs = self.R.uniform(low=self.frequencies[0], high=self.frequencies[1])
+
+        length = signal.shape[len(signal.shape) - 1]
+
+        time_partial = np.arange(0, round(self.fracs * length), 1)
+        squaredpulse_partial = self.magnitude * squarepulse(self.freqs * time_partial)
+
+        loc = np.random.choice(range(length))
+        signal = paste(signal, squaredpulse_partial, (loc,))
+
+        return signal
+
+
+class SignalFillEmpty(Transform):
+    """
+    replace empty part of a signal (NaN)
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(self, replacement: float = 0.0) -> None:
+        """
+        Args:
+            replacement: value to replace nan items in signal
+        """
+        super().__init__()
+        self.replacement = replacement
+
+    def __call__(self, signal: NdarrayOrTensor) -> NdarrayOrTensor:
+        """
+        Args:
+            signal: signal to be filled
+        """
+        signal = torch.nan_to_num(convert_to_tensor(signal), nan=self.replacement)
```

Comment:
this `torch.nan_to_num` is not available in torch 1.7, perhaps 
```py
======================================================================
ERROR: test_correct_parameters_multi_channels (tests.test_signal_fillempty.TestSignalRandDrop)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/MONAI/MONAI/tests/test_signal_fillempty.py", line 28, in test_correct_parameters_multi_channels
    fillemptysignal = fillempty(sig)
  File "/home/runner/work/MONAI/MONAI/monai/transforms/signal/array.py", line 381, in __call__
    signal = torch.nan_to_num(convert_to_tensor(signal), nan=self.replacement)
AttributeError: module 'torch' has no attribute 'nan_to_num'

----------------------------------------------------------------------
```

you can put a decorator in the test to skip the case, for example: https://github.com/Project-MONAI/MONAI/blob/1516ca758090dd8ed2890b64554a77f681a0e224/tests/test_reg_loss_integration.py#L43-L45

Reply:
it seems `torch.pi` is also not available in the previous versions https://github.com/Project-MONAI/MONAI/runs/7920616362?check_suite_focus=true